STF to fort90 converter

kyrsjo opened the pull request:
New PR, now from the right repository.

vkbo commented:
When will this be finished?

kyrsjo commented on test/CMakeLists.txt:
```diff
@@ -29,13 +28,24 @@ set_tests_properties(SelfTest_read90 PROPERTIES PASS_REGULAR_EXPRESSION "Error i
+add_custom_command(TARGET test_copy_SelfTest_splitSingletrack_run POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/test/CheckSplitSingletrack/CheckSplitSingletrack.sh ${CMAKE_BINARY_DIR}/test/CheckSplitSingletrack/CheckSplitSingletrack.sh)
+add_dependencies(SixTrackCore test_copy_SelfTest_splitSingletrack_run)
+
+add_test(NAME SelfTest_splitSingletrack_run WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/test/CheckSplitSingletrack/ COMMAND ${CMAKE_BINARY_DIR}/test/CheckSplitSingletrack/CheckSplitSingletrack.sh ${CMAKE_BINARY_DIR}/splitSingletrack ${CMAKE_BINARY_DIR}/read90 ${CMAKE_BINARY_DIR}/test/dump7_fma/singletrackfile.dat.canonical)
```
TODO: This hard-codes the use of the `dump7_fma` canonical singletrackfile.

kyrsjo commented:
Done!

vkbo commented on source/splitSingletrack/splitSingletrack.f90:
```diff
@@ -0,0 +1,73 @@
+    call get_command_argument(cmdarg_i, cmdarg_arg, cmdarg_length, cmdarg_status)
+    if (len_trim(cmdarg_arg) == 0) exit
+
+    if (cmdarg_i .eq. 0) then
```
Please don't use a mix of fortran 77 and newer logical operators. Should be `==`.

vkbo commented on source/splitSingletrack/mod_splitSingletrack.f90:
```diff
@@ -0,0 +1,210 @@
+      else
+        write(ofnameBuffer,'(a,a,i0.6)') ifname,'.',iPair
+      end if
+      !write(*,*)ofnameBuffer
```
Forgot a debug line here ...

vkbo commented on source/splitSingletrack/mod_splitSingletrack.f90:
```diff
@@ -0,0 +1,210 @@
+    stop 1
+
+    !close (inUnit)
+    !close (outUnit)
```
Debug code?

vkbo commented on source/splitSingletrack/mod_splitSingletrack.f90:
```diff
@@ -0,0 +1,210 @@
+  integer :: inUnit = 90
+  integer :: outUnit = 91
+
+  integer :: myOutUnit = output_unit
```
Why use `*` in the program and `output_unit` here? Should be consistent. I think `*` is fine anyway. `my` prefix is anyway only for example code, not for production.

kyrsjo replied:
The output_unit variable is used in the module so that it is easyer to integrate with the top-level SixTrack. I could get the iso_fortran_env::output unit in the program and use that (and set it for the module), but it would not make any difference.

I'll change the variable name.

vkbo replied:
Would probably make more sense to just call it lout then, but ok.

vkbo commented on source/splitSingletrack/mod_splitSingletrack.f90:
```diff
@@ -0,0 +1,210 @@
+         dnumlr,sigcor,dpscor
+    close(inUnit)
+
+    if (modulo(itopa,2) /= 0) then
```
Should first check that `itopa` > 0. Also, mod and modulo behaves differently, but maybe not for integers.

kyrsjo replied:
AFAIK the difference only shows up if one of the numbers are negative, which can't happen.

I added the check (which would really be a case of corrupted file).

kyrsjo commented:
Thanks for your comments, implemented what made sense. I also added a footnote to the manual mentioning the new tool -- it could probably be better integrated in the text, but at least it's there.
